Sources/NIOHTTP2/HTTP2StreamChannel.swift

```diff
@@ -554,11 +554,27 @@ private extension HTTP2StreamChannel {
         self.pendingReads = CircularBuffer(initialCapacity: 0)
     }
 
-    /// DinitialCapacityreads to the channel.
+    /// Deliver all pending reads to the channel.
     private func deliverPendingReads() {
         assert(self.isActive)
         while self.pendingReads.count > 0 {
-            self.pipeline.fireChannelRead(NIOAny(self.pendingReads.removeFirst()))
+            let frame = self.pendingReads.removeFirst()
+
+            let flowControlledFameSize: Int?
+            if case .data(let data) = frame.payload {
+                flowControlledFameSize = data.data.readableBytes
+            } else {
+                flowControlledFameSize = nil
```

Comment:
We should not call this the flow-controlled frame size because it isn't: at this stage we don't know what the padding was. This doesn't actually _matter_ here, it's a good thing, but we should just say "dataLength" or something similar.